contrib/epee/include/readline_buffer.h

```diff
@@ -19,12 +21,23 @@ namespace rdln
     }
     void get_line(std::string& line) const;
     void set_prompt(const std::string& prompt);
+    static void add_completion(const std::string& command)
+    {
+      if(std::find(completion_commands.begin(), completion_commands.end(), command) != completion_commands.end())
+	return;
+      completion_commands.push_back(command);
```

Comment:
> Now that I expanded the code I see that this is actually duplicating an existing std::map

Where?

> And std::sort + std::lexicographical_compare + std::lower_bound would be interesting but not particularly noticeable...

Really not sure where you're going with this. The code is simple, low overhead, easy to read addition to implement readline tab completion of commands. Is there an edit you are suggesting?